src/ops/crate_spec.rs

```diff
@@ -0,0 +1,40 @@
+use anyhow::Context as _;
+use cargo::{util::validate_package_name, CargoResult};
+
+/// User-specified crate
+///
+/// This can be a
+/// - Name (e.g. `docopt`)
+/// - Name and a version req (e.g. `docopt@^0.8`)
+/// - Path
+#[derive(Debug)]
+pub struct CrateSpec {
```

Comment:
If you are already depending on cargo, is there a reason not to use `cargo::core::PackageIdSpec`?

Reply:
yeah. I can reuse it. 😆

Reply:
I have a question about this code: What does it mean by `- Path`? It seems we don't support the path in it.

Reply:
Ah, copied from cargo-add?

It used to auto-detect between package names and paths.  That was stripped out but apparently the comment was never updated.